File: game-manager-jk-server-connector/src/main/java/com/gamemanager/JediAcademyServerConnector.java
package com.gamemanager;

import lombok.AllArgsConstructor;
import lombok.Getter;

import java.io.IOException;
import java.net.DatagramPacket;
import java.net.DatagramSocket;
import java.net.InetAddress;
import java.net.UnknownHostException;

@AllArgsConstructor
@Getter
public class JediAcademyServerConnector {
	
	private final InetAddress inetAddress;
	private final int port;
	private static final byte OBB = (byte) 0xff;
	
	private static final int TIMEOUT_MILLISECONDS = 3000; // 10 sec
	private static final int MAX_PACKET_SIZE = 65507;
	private static final String OBB_HEADER_PLACEHOLDER = "xxxx";
	
	public JediAcademyServerConnector(String host,
									int port) throws UnknownHostException {
		this.inetAddress = InetAddress.getByName(host);
		this.port = port;
	}
	
	public byte[] executeCommand(String input) throws IOException {
		return doRequest(formatInput(input));
	}
	
	private byte[] doRequest(byte[] inputBytes) throws IOException {
		final byte[] outputBytes = new byte[MAX_PACKET_SIZE];
		
		try (DatagramSocket socket = new DatagramSocket()) {
			socket.setSoTimeout(TIMEOUT_MILLISECONDS);
			
			final DatagramPacket packetToSend = new DatagramPacket(inputBytes, inputBytes.length, inetAddress, port);
			socket.send(packetToSend);
			
			final DatagramPacket packetToReceive = new DatagramPacket(outputBytes, outputBytes.length);
			socket.receive(packetToReceive);
		}
		
		return outputBytes;
	}
	
	private byte[] insertObbHeader(byte[] inputBytes) {
		inputBytes[0] = OBB;
		inputBytes[1] = OBB;
		inputBytes[2] = OBB;
		inputBytes[3] = OBB;
		
		return inputBytes;
	}
	
	private byte[] formatInput(String input) {
		return insertObbHeader((OBB_HEADER_PLACEHOLDER + input).getBytes());
	}
	
}

File: game-manager-jk-server-connector/src/main/java/com/gamemanager/JediAcademyServerConnector.kt
package com.gamemanager

import java.io.IOException
import java.net.DatagramPacket
import java.net.DatagramSocket
import java.net.InetAddress

open class JediAcademyServerConnector(
		val inetAddress: InetAddress,
		val port: Int
) {
	constructor(host: String, port: Int) : this(InetAddress.getByName(host), port)

	private val OBB = 0xff.toByte()
	private val TIMEOUT_MILLISECONDS = 10000 // 10 sec
	private val MAX_PACKET_SIZE = 65507
	private val OBB_HEADER_PLACEHOLDER = "xxxx"

	@Throws(IOException::class)
	fun executeCommand(input: String): ByteArray {
		return doRequest(formatInput(input))
	}

	@Throws(IOException::class)
	private fun doRequest(inputBytes: ByteArray): ByteArray {
		val outputBytes = ByteArray(MAX_PACKET_SIZE)

		DatagramSocket().use {

			it.soTimeout = TIMEOUT_MILLISECONDS

			val packetToSend = DatagramPacket(inputBytes, inputBytes.size, inetAddress, port)
			it.send(packetToSend)

			val packetToReceive = DatagramPacket(outputBytes, outputBytes.size)
			it.receive(packetToReceive)
		}

		return outputBytes
	}

	private fun insertObbHeader(inputBytes: ByteArray): ByteArray {

		inputBytes[0] = OBB
		inputBytes[1] = OBB
		inputBytes[2] = OBB
		inputBytes[3] = OBB

		return inputBytes
	}

	private fun formatInput(input: String): ByteArray {
		return insertObbHeader((OBB_HEADER_PLACEHOLDER + input).toByteArray())
	}


}